File: SRC/Host/MyTopLevelComponent.vb
Imports System.ComponentModel
Imports System.ComponentModel.Design

Namespace Host
    ''' <summary>
    ''' Uses the custom RootDesigner (MyRootDesigner)
    ''' </summary>
    <Designer(GetType(MyRootDesigner), GetType(IRootDesigner))>
    <Designer(GetType(ComponentDesigner))>
    Public Class MyTopLevelComponent
        Inherits Component

        Public Sub New()
        End Sub
    End Class
End Namespace ' namespace

File: SRC/Host/MyTopLevelComponent.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Windows.Forms;
using System.Data;
using System.Windows.Forms.Design;
using System.Drawing.Design;

namespace Host
{
    /// <summary>
    /// Uses the custom RootDesigner (MyRootDesigner)
    /// </summary>
    [Designer(typeof(MyRootDesigner), typeof(IRootDesigner))]
	[Designer(typeof(ComponentDesigner))]
	public class MyTopLevelComponent : Component
	{
        public MyTopLevelComponent()
		{
		}
	}

}// namespace
